session/generate.go

```diff
@@ -0,0 +1,24 @@
+package session
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+)
+
+type SessionId string
+
+func GenerateSessionId() (sid SessionId, err error) {
+	b, err := generateRandomBytes(16)
```

Comment:
Did You consider using UniqueID library, which has randomness issues solved?
I used https://github.com/satori/go.uuid:
```
requestId := uuid.NewV4().String()
```